build/install.bzl

```diff
@@ -0,0 +1,62 @@
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Installs a file to the actual workspace outside of the sandbox.
+"""
+
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+def _install_impl(ctx):
+
+    cmd = ""
```

Comment:
An alternative implementation to consider:

```python
def _install_impl(ctx):
    copyFormat = "cp -f {} {}"
    touchFormat = "touch {}"
    cmdList = []

    for s in ctx.attr.srcs:
        files = s.files.to_list()
        file = files[0].path
        root_file = paths.relativize(file, "bazel-out/k8-fastbuild/genfiles/")
        cmdList.append(copyFormat.format(file, root_file))

    cmdList.append(touchFormat.format(ctx.outputs.record.path))
    cmd = " && ".join(cmdList)
    ...
```

If you want to be really safe, the last line should be:

```python
# Remove empty items from cmdList to avoid an erroneous " && && "
" && ".join(filter(None, cmdList))
```

I'm also looking into how to make this work on a Darwin system. It's failing for me with this error:

```
 salazar:cluster-api-provider-aws cha$ make crds
bazel build //config:config_install
WARNING: failed to raise resource limit 8 to 524288: Invalid argument
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 456b2a2b-7895-4f8f-865b-45226a773acc
ERROR: /Users/cha/go/src/sigs.k8s.io/cluster-api-provider-aws/config/BUILD.bazel:17:1: in install rule //config:config_install: 
Traceback (most recent call last):
	File "/Users/cha/go/src/sigs.k8s.io/cluster-api-provider-aws/config/BUILD.bazel", line 17
		install(name = 'config_install')
	File "/Users/cha/go/src/sigs.k8s.io/cluster-api-provider-aws/build/install.bzl", line 28, in _install_impl
		paths.relativize(file, "bazel-out/k8-fastbuild/genf...")
	File "/private/var/tmp/_bazel_cha/b7ce3b297716e241b109da3b2c60846f/external/bazel_skylib/lib/paths.bzl", line 186, in paths.relativize
		fail(("Path '%s' is not beneath '%s'"...)))
Path 'bazel-out/darwin-fastbuild/genfiles/config/rbac/rbac_role.yaml' is not beneath 'bazel-out/k8-fastbuild/genfiles/'
ERROR: Analysis of target '//config:config_install' failed; build aborted: Analysis of target '//config:config_install' failed; build aborted
INFO: Elapsed time: 13.016s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (358 packages loaded, 9342 targets configured)
make: *** [crds] Error 1
```

Reply:
Thanks for checking this out, was interested how it'd behave on os x. Will dig deeper, or more likely, split it out into a separate PR